extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayCreator.java

```diff
@@ -33,7 +33,21 @@ public FlywayCreator withCallbacks(Collection<Callback> callbacks) {
 
     public Flyway createFlyway(DataSource dataSource) {
         FluentConfiguration configure = Flyway.configure();
-        configure.dataSource(dataSource);
+
+        if (flywayRuntimeConfig.jdbcUrl.isPresent()) {
+            if (flywayRuntimeConfig.username.isPresent() && flywayRuntimeConfig.password.isPresent()) {
+                configure.dataSource(flywayRuntimeConfig.jdbcUrl.get(), flywayRuntimeConfig.username.get(),
+                        flywayRuntimeConfig.password.get());
+            }
+        } else {
+            if (flywayRuntimeConfig.username.isPresent() && flywayRuntimeConfig.password.isPresent()) {
+                configure.dataSource(
+                        new CredentialsDataSource(dataSource, flywayRuntimeConfig.username.get(),
+                                flywayRuntimeConfig.password.get()));
+            } else {
```

Comment:
Sure, please give me some time to work on it.

Reply:
@sberyozkin  I'm get this error below when I try to use CrednetialsProviderFinder.find

> 
> [ERROR] io.quarkus.flyway.runtime.FlywayCreatorTest.testIgnoreMissingMigrations  Time elapsed: 0.022 s  <<< ERROR!
> java.lang.NullPointerException: Cannot invoke "io.quarkus.arc.ArcContainer.instance(java.lang.Class, java.lang.annotation.Annotation[])" because "container" is null
> 	at io.quarkus.credentials.runtime.CredentialsProviderFinder.find(CredentialsProviderFinder.java:13)
> 	at io.quarkus.flyway.runtime.FlywayCreator.createFlyway(FlywayCreator.java:48)
> 	at io.quarkus.flyway.runtime.FlywayCreatorTest.createdFlywayConfig(FlywayCreatorTest.java:285)
> 	at io.quarkus.flyway.runtime.FlywayCreatorTest.testIgnoreMissingMigrations(FlywayCreatorTest.java

:206

Could you give me a explanation how I should implement this?

Thank you !

Reply:
Hi,
I added the code below, but i'm facing problem above with ArcContainer ? I think there is more steps to get the container ready, could you give some help explain that? @sberyozkin 

>    String beanName = flywayRuntimeConfig.credentialsProviderName.orElse(null);
>             CredentialsProvider credentialsProvider = CredentialsProviderFinder.find(beanName);
>             Map<String, String> mapCredentials = credentialsProvider.getCredentials(null);
>             if (mapCredentials.get(CredentialsProvider.USER_PROPERTY_NAME) != null &&
>                     mapCredentials.get(CredentialsProvider.PASSWORD_PROPERTY_NAME) != null) {
>                 configure.dataSource(
>                         new CredentialsDataSource(dataSource, mapCredentials.get(CredentialsProvider.USER_PROPERTY_NAME),
>                                 mapCredentials.get(CredentialsProvider.PASSWORD_PROPERTY_NAME)));

Reply:
I think this can be implemented in a separate PR once there is a demand for this use case. For now, I'd recommend keeping it simple for now.

Reply:
Sorry for a delay, @mfpc and thanks for giving it a try :+1:  
Thanks @gastaldi, makes sense